Algorithm.CSharp/FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm.cs

```diff
@@ -0,0 +1,129 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+using QuantConnect.Interfaces;
+using QuantConnect.Data.Market;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Continuous Futures Regression algorithm. 
+    /// Asserting the behavior of stop market order <see cref="StopMarketOrder"/> in extended market hours 
+    /// <seealso cref="Data.UniverseSelection.UniverseSettings.ExtendedMarketHours"/>
+    /// </summary>
+    public class FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private OrderTicket _ticket;
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 6);
+            SetEndDate(2013, 10, 12);
+
+            var future = AddFuture(Futures.Indices.SP500EMini, Resolution.Minute, extendedMarketHours: true);
+
+            Schedule.On(DateRules.EveryDay(), TimeRules.At(19, 0), () =>
+            {
+                MarketOrder(future.Mapped, 1);
+                _ticket = StopMarketOrder(future.Mapped, -1, future.Price * 0.999m);
```

Comment:
Should assert that this order is filled on extended hours, perhaps in `OnOrderEvent`

Reply:
Added:
- UTest
- regression algos